hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/ContractCreateTransactionHandler.java

```diff
@@ -116,4 +124,64 @@ public void updateContractResult(ContractResult contractResult, RecordItem recor
             contractResult.setGasLimit(transactionBody.getGas());
         }
     }
+
+    public void updateChildFromParent(Contract contract, RecordItem recordItem) {
+        if (!entityProperties.getPersist().isContracts() || !recordItem.isChild() || recordItem.getParent() == null) {
```

Comment:
can be simplified to 
```suggestion
        if (recordItem.getParent() == null) {
```

`isContracts()` is checked in the caller

Reply:
Yeah, I dropped one of the tests after realizing this.
Conceptually I see why it can be reduced further, however `recordItem.isChild()` is set as a result of transactionRecordBytes parsing, whereas `recordItem.getParent()` is set as a result of comparison logic between current and previous.
Because the logic exists in different flows I think it's fair to have the explicit checks.
Let me know if you disagree

Reply:
I'd say checking parent set or not is enough, since if it's not a child there definitely shouldn't be parent set. if in any chance the parent is set, that's a bug in `RecordItem`, not here.